src/SqlAsyncCollector.cs

```diff
@@ -95,7 +96,19 @@ public async Task AddAsync(T item, CancellationToken cancellationToken = default
                 TelemetryInstance.TrackEvent(TelemetryEventName.AddAsync);
                 try
                 {
-                    this._rows.Add(item);
+                    if (item is string)
+                    {
+                        this._rows.Add(JsonConvert.DeserializeObject<T>(item as string));
+                    }
+                    else if (item.GetType() == typeof(JObject))
+                    {
+                        dynamic jsonString = JObject.Parse(item.ToString());
```

Comment:
Why is this dynamic? And the name `jsonString` implies it's converting it to a string but `JObject.Parse` returns a JObject. 

Are you sure this shouldn't just be

`this._rows.Add(item.ToString());`

?

Reply:
I did that and compiler complained -> Argument 1: cannot convert from 'string' to 'T' So I used dynamic.

Reply:
This isn't even a string though. You're parsing an object - so `jsonString` is actually a JObject after this...

Reply:
renamed it as itemAsJObject, hope that helps.